### PR TITLE
Fix Flow leak when has .one() step and is only Flow emits rxDismissed #trivial

### PR DESCRIPTION
**Pull request by asiliuk**
### This leak can happen in following situation:
* Coordinator leaves longer then presented Flow, for example as `let` on not yet migrated ViewModel who is presenting already migrated to Flow components
* You present modally flow whose `root` is UINavigationController
* It returns .one() with child `Presentable` and puts its ViewController inside root.viewControllers
* User swipes to dismiss whole modally presented `UINavigationViewController`

### What is happening in that case:

1) Both UINavigationController (Flow implementing class) and child Presentable receive `didDisappear`

2) Because of implementation of `rx.dismissed` on UIViewController event on child presentable is filtered 
```swift
.filter { [weak base] _ in base?.isBeingDismissed ?? true }
```
And `isBeingDismissed` of a child is `false` for some reason

3) In `func coordinate(...)` of FlowCoordinator there's this code
```swift
.flatMap { [weak self] in
    self?.steps(from: $0, within: flow, allowStepWhenDismissed: allowStepWhenDismissed) ?? Signal.empty()
}
```
Since rxDimiss of child Presentable is filtered out this code is equivalent of
```
.flatMap { [weak self] _ in
    .never()
}
```

4) So first part of pipe in `func coordinate()` is destroyed because Flow rxDismissed emitted event
5) Bottom part leaves forever and is disposed only when FlowCoordinator with its disposeBag is destroyed
6) And because FlowCoordinator in some cases can leave longer that Flow that it presents it can cause unexpected leaks

Basically fix is to destroy whole pipe when presented Flow is dimissed

## Checklist

- [X] this PR is based on develop or a 'develop related' branch
- [X] the commits inside this PR have explicit commit messages
- [ ] the Jazzy documentation has been generated (if needed -> Jazzy RxFlow)

### CODE REVIEW

**Comment by asiliuk** [comment]
@twittemb Hi! Do you have any thoughts about this?

**Comment by twittemb** [comment]
Sorry guys, I’ll review it ASAP I promise (before the end of the week).

**Review by twittemb — APPROVED** [review]

**Comment by twittemb** [comment]
Thanks a lot
